api-authorization/server/authorization-middleware.js

```diff
@@ -0,0 +1,32 @@
+const jwt = require('jsonwebtoken'); // eslint-disable-line
+const ClientError = require('./client-error'); // eslint-disable-line
+
+function authorizationMiddleware(req, res, next) {
+  /* your code here */
+  if (!req.headers['x-access-token']) {
+    throw new ClientError(401, 'athentication required');
+  }
+  const token = req.headers['x-access-token'];
+  try {
+    req.user = jwt.verify(token, process.env.TOKEN_SECRET);
+  } catch (err) {
+    console.log(err); //eslint-disable-line
+  }
```

Comment:
If an error happens, you should pass it to `next(err)`. Alternatively, because this function is not asynchronous, you can leave out the `try/catch` block, and Express will send the error to your `errorHandler` for you.